compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -1000,6 +1001,51 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
         err.emit()
     }
 
+    fn apply_do_not_recommend<'b>(
+        &self,
+        trait_predicate: &'_ mut ty::Binder<'b, ty::TraitPredicate<'tcx>>,
```

Comment:
This `'b` lifetime is unnecessary. It should always be `'tcx`. If not, then there's almost certainly some other lifetime in the caller that needs to be turned into a `'tcx`.